src/common/services/tokenApi.js

```diff
@@ -29,15 +29,12 @@ module.exports = {
       }, (error, response, body) => {
         if (error) {
           logger.error('There was a problem calling the settoken API');
-          return console.dir(error);
+          logger.error(error);
+          reject(error);
         }
         logger.debug('Successfully called settoken API');
         resolve(body);
-        return body;
       });
-    }).catch((err) => {
```

Comment:
This catch block does not get triggered, as the code just returns a Promise, not invoke it.